packages/flutter/lib/src/cupertino/context_menu.dart

```diff
@@ -0,0 +1,1224 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+import 'dart:ui' as ui;
+import 'package:flutter/gestures.dart' show kMinFlingVelocity, kLongPressTimeout;
+import 'package:flutter/rendering.dart';
+import 'package:flutter/scheduler.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter/widgets.dart';
+import 'context_menu_action.dart';
+
+// The scale of the child at the time that the ContextMenu opens.
+// This value was eyeballed from a physical device running iOS 13.1.2.
+const double _kOpenScale = 1.05;
+
+typedef _DismissCallback = void Function(
+  BuildContext context,
+  double scale,
+  double opacity,
+);
+
+// Given a GlobalKey, return the Rect of the corresponding RenderBox's
+// paintBounds in global coordinates.
+Rect _getRect(GlobalKey globalKey) {
+  assert(globalKey.currentContext != null);
+  final RenderBox renderBoxContainer = globalKey.currentContext.findRenderObject();
+  final Offset containerOffset = renderBoxContainer.localToGlobal(renderBoxContainer.paintBounds.topLeft);
+  return containerOffset & renderBoxContainer.paintBounds.size;
+}
+
+// The context menu arranges itself slightly differently based on the location
+// on the screen of [ContextMenu.child] before the [ContextMenu] opens.
+enum _ContextMenuLocation {
+  center,
+  left,
+  right,
+}
+
+/// A full-screen modal route that opens when the [child] is long-pressed.
+///
+/// When open, the ContextMenu shows the child, or [preview] if given, in a
```

Comment:
nit: `ContextMenu` can be put in square brackets (or should it?), so are `Overlay`, `Expanded` and `PopupRoute`.

Reply:
Done, but with a caveat.  I left ContextMenu unbracketed because this is the definition of ContextMenu itself, so the link would go to the same page.  Is that correct, or should I still bracket it?

Reply:
I asked the same question in different PR. @goderbauer's answer was we should still square bracket it.

Reply:
Alright I'm bracketing that too.

Also, should I only bracket the first occurrence?  That's what I've been doing to avoid having lots of duplicate links right next to each other.